src/platform/plugins/shared/data_views/common/fields/field_list.ts

```diff
@@ -28,6 +28,12 @@ export interface IIndexPatternFieldList extends Array<DataViewField> {
    * @returns a new data view field instance
    */
   create(field: FieldSpec): DataViewField;
+
+  /**
```

Comment:
Built on top of this PR: https://github.com/elastic/kibana/pull/224767 to test performance